### PR TITLE
Use Gecko atoms in geckolib.

### PR DESCRIPTION
**Pull request by Ms2ger**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11095)

### CODE REVIEW

**Comment by Ms2ger** [comment]
r? @bholley 

This still needs the corresponding changes to gecko.

File: components/style/properties/longhand/font.mako.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
         #[derive(Debug, PartialEq, Eq, Clone, Hash, HeapSizeOf, Deserialize, Serialize)]
         pub enum FontFamily {
-            FamilyName(Atom),
+            FamilyName(String),
```
Comment (bholley) [inline components/style/properties/longhand/font.mako.rs]: What's the reason for this changeset?

File: components/style/restyle_hints.rs [hunk 2]
```diff
@@ -161,7 +161,7 @@ impl<'a, E> Element for ElementWrapper<'a, E>
                 match attr.namespace {
                     NamespaceConstraint::Specific(ref ns) => self.snapshot.get_attr(ns, local_name),
                     NamespaceConstraint::Any => self.snapshot.get_attr_ignore_ns(local_name),
-                }.map_or(false, |v| test(v))
+                }.map_or(false, |v| test(&v.to_string()))
```
Comment (bholley) [inline components/style/restyle_hints.rs]: What does the to_string implementation in string_cache look like? Does it copy? If so, I'm not sure if it's acceptable to use here.

File: components/style/properties/longhand/font.mako.rs (hunk 1 above)
Comment (Ms2ger) [inline components/style/properties/longhand/font.mako.rs]: Gecko atoms don't have a utf-8 representation, so it's impossible to support `FontFamily::name()` with it.

File: components/style/restyle_hints.rs (hunk 2 above)
Comment (Ms2ger) [inline components/style/restyle_hints.rs]: It does copy. The alternative is storing a `DOMString` in `AttrValue::Atom` along with the `Atom`.

**Comment by KiChjang** [comment]
```
/home/travis/build/servo/servo/components/script/dom/attr.rs:252:13: 252:37 error: this pattern has 1 field, but the corresponding variant has 2 fields [E0023]

/home/travis/build/servo/servo/components/script/dom/attr.rs:252             AttrValue::Atom(ref val) => Some(val.clone()),

                                                                             ^~~~~~~~~~~~~~~~~~~~~~~~

/home/travis/build/servo/servo/components/script/dom/attr.rs:252:13: 252:37 help: run `rustc --explain E0023` to see a detailed explanation
```

**Comment by bholley** [comment]
I'll take the Atom stuff from here - thanks for the help!

**Pull request closed without merging**